modules/transactions/src/main/java/org/apache/ignite/internal/tx/LockKey.java

```diff
@@ -73,6 +75,12 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
+        // Apply more efficient hashing to byte buffers to decrease collisions
+        if (key instanceof ByteBuffer) {
+            ByteBuffer key1 = (ByteBuffer) key;
+            return HashUtils.hash32(HashUtils.hash64(key1, 0, key1.capacity(), contextId != null ? contextId.hashCode() : 0));
```

Comment:
hash32 calls hash64 internally 
```
    public static int hash32(long data, int seed) {
        long hash = hash64(data, seed);

        return (int) (hash ^ (hash >>> 32));
    }
```

Reply:
What is wrong here? I need to get 32 bit hash as a result of mixing byte array hash with a seed.